src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -44,6 +45,11 @@ public TelemetryClient(TelemetryConfiguration configuration)
             }
 
             this.configuration = configuration;
+
+            if (this.configuration.TelemetryChannel == null)
```

Comment:
Since the client no longer creates one if it is not provided I thought it might be nicer to have a meaningful message rather than a NullReferenceException.